engine/access/access_test.go

```diff
@@ -107,6 +131,15 @@ func (suite *Suite) SetupTest() {
 
 	suite.chainID = flow.Testnet
 	suite.metrics = metrics.NewNoopCollector()
+
+	suite.finalizationDistributor = pubsub.NewFinalizationDistributor()
+	suite.finalizedHeaderCache, _ = synceng.NewFinalizedHeaderCache(suite.log, suite.state, suite.finalizationDistributor)
+
+	unittest.RequireCloseBefore(suite.T(), suite.finalizedHeaderCache.Ready(), time.Second, "expect to start before timeout")
+}
+
+func (suite *Suite) TearDownTest() {
+	unittest.RequireCloseBefore(suite.T(), suite.finalizedHeaderCache.Done(), time.Second, "expect to stop before timeout")
```

Comment:
given the previous change, it's possible `suite.finalizedHeaderCache` isn't set, so we should gate the shutdown or the suite will crash
```suggestion
	if suite.finalizedHeaderCache != nil {
		unittest.RequireCloseBefore(suite.T(), suite.finalizedHeaderCache.Done(), time.Second, "expect to stop before timeout")
	}
```